stripped/colabs/agregacje_kilkuletnie_stripped.py

```diff
@@ -0,0 +1,669 @@
+from holoviews import opts
+from IPython.display import IFrame
+from IPython.core.display import display
+from bokeh.plotting import show, output_notebook
+from sklearn import preprocessing
+from sklearn.mixture import GaussianMixture
+from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, jaccard_score, recall_score, f1_score
+import pickle
+import seaborn as sns
+from sklearn import tree
+from sklearn.metrics import confusion_matrix, classification_report
+from sklearn.model_selection import train_test_split
+from sklearn.feature_selection import mutual_info_classif
+from typing import Optional, Tuple, List
+from imblearn.over_sampling import RandomOverSampler, SMOTE
+from bokeh.layouts import row
+from pyspark.sql.functions import col
+from pyspark.sql.functions import size
+from pyspark.sql import SparkSession
+from pyspark.sql.types import IntegerType, FloatType, StringType, StructType
+from pyspark.sql import Window
+import pyspark
+from sklearn.ensemble import RandomForestClassifier
+import lightgbm as lgb
+
+"""# Załadowanie zbioru danych"""
+
+from google.colab import drive
+drive.mount("/content/drive")
+
+!git clone https://github.com/PiotrMaciejKowalski/BigData2024Project.git
+
+!chmod 755 /content/BigData2024Project/src/setup.sh
+!/content/BigData2024Project/src/setup.sh
+
+import sys
+sys.path.append('/content/BigData2024Project/src')
+
+from start_spark import initialize_spark
+initialize_spark()
+
+from pyspark.sql import SparkSession
+from big_mess.loaders import default_loader
+from pyspark.sql import SparkSession, DataFrame as SparkDataFrame, functions as F
+
+spark = SparkSession.builder\
+        .master("local")\
+        .appName("Colab")\
+        .config('spark.ui.port', '4050')\
+        .getOrCreate()
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# nasa = default_loader(spark)
+# nasa.show(5)
+
+nasa.createOrReplaceTempView("nasa")
+
+"""# Średnie wieloletnie dla poszczególnych miesięcy"""
+
+def overYearMonthStats(columns: list, n: int):
+    """
+    Funkcja licząca średnie dla wybranych zmiennych dla danych współrzędnych geograficznych
+    dla każdego z 12 miesięcy od roku w rozważanym pomiarze do n lat wstecz (tj. gdyby n=3, to
+    średnia zostałaby wyliczona z rozważanego roku oraz 3 lat przed nim, czyli w praktyce mielibyśmy
+    średnią z łącznie 4 lat).
+
+    :param columns: lista nazw rozważanych kolumn ze zbioru danych
+    :param n: liczba lat wstecz, z których chcemy wyliczyć średnią
+    """
+
+    if not columns:
+      raise ValueError("Podana musi zostać co najmniej jedna zmienna")
+
+    windowSpec = Window.partitionBy("lon", "lat", "Month").orderBy("Year").rowsBetween(-n, 0)
+
+    nasa_window = nasa
+    for column in columns:
+        average_column_name = "monthly_avg_" + str(n) + "years_" + column
+        nasa_window = nasa_window.withColumn(average_column_name, F.avg(F.col(column)).over(windowSpec))
+
+    return nasa_window
```

Comment:
Jeżeli funkcja coś zwraca to może warto dodać typing "def overYearMonthStats(columns: list, n: int) -> ..."

Reply:
zrobione